### ISSUE TITLE
`no-unused-expressions` false positive in `expect().to.be.true` for chai

### ISSUE THREAD

**Issue by aminya** [eslint/eslint#14213]
**Tell us about your environment**

* **ESLint Version:** ^7.18.0
* **Node Version:** v15.11.0
* **npm Version:** 6.14.11 
* **Operating System:**  Win 10

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

I am using [eslint-config-atomic](https://github.com/atom-community/eslint-config-atomic/), which uses `@typescript-eslint/parser`

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "extends": "eslint-config-atomic/strict",
  "env": {
      "mocha": true,
      "jasmine": false
  },
  "ignorePatterns": ["build/", "node_modules/"]
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
describe("test", () => {
    it("test", () => {
      expect([1,2].some((e) => e === 1)).to.be.true
	})
})
```

```bash
eslint .
```

**What did you expect to happen?**

This expression is used, but 

**What actually happened? Please copy-paste the actual, raw output from ESLint.**

**Steps to reproduce this issue:**

Reproduction
https://eslint.org/demo#eyJ0ZXh0IjoiZGVzY3JpYmUoXCJ0ZXN0XCIsICgpID0+IHtcbiAgICBpdChcInRlc3RcIiwgKCkgPT4ge1xuICAgICAgZXhwZWN0KFsxLDJdLnNvbWUoKGUpID0+IGUgPT09IDEpKS50by5iZS50cnVlXG5cdH0pXG59KSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTIsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnsianN4Ijp0cnVlLCJnbG9iYWxSZXR1cm4iOnRydWUsImltcGxpZWRTdHJpY3QiOnRydWV9fSwicnVsZXMiOnsibm8tdW51c2VkLWV4cHJlc3Npb25zIjoyfSwiZW52Ijp7ImJyb3dzZXIiOmZhbHNlLCJub2RlIjpmYWxzZSwibW9jaGEiOnRydWV9fX0=

**Are you willing to submit a pull request to fix this bug?**

**Comment by mdjermanovic** [comment]
Hi @aminya, thanks for the issue!

This is intended behavior. This rule doesn't account for exceptional cases when member expression can trigger side effects. I don't think the rule should allow all member expressions by default, and it can't be aware of library-specific features like those of Chai's `expect`. We could consider adding a generic option if there's is a concrete proposal to evaluate (please feel free to open a rule change request issue). An alternative is [eslint-plugin-chai-friendly](https://www.npmjs.com/package/eslint-plugin-chai-friendly). 

I'm closing this issue since it isn't a bug.

**Comment by aminya** [comment]
Oh, thanks!

eslint-plugin-chai-friendly works perfectly. I didn't know about that. This issue would be a good reference in the future.
